File: src/main/java/org/noxet/noxetserver/playerstate/properties/PSPExperienceLevel.java
package org.noxet.noxetserver.playerstate.properties;

import org.bukkit.entity.Player;
import org.noxet.noxetserver.playerstate.PlayerStateProperty;

public class PSPExperienceLevel implements PlayerStateProperty<Integer> {
    @Override
    public String getConfigName() {
        return "experience_level";
    }

    @Override
    public Integer getDefaultSerializedProperty() {
        return 0;
    }

    @Override
    public Integer getSerializedPropertyFromPlayer(Player player) {
        return player.getLevel();
    }

    @Override
    public void restoreProperty(Player player, Integer level) {
        player.setLevel(level);
    }
}

File: src/main/java/org/noxet/noxetserver/playerstate/properties/PSPExperienceLevel.scala
package org.noxet.noxetserver.playerstate.properties

import org.bukkit.entity.Player
import org.noxet.noxetserver.playerstate.PlayerStateProperty

object PSPExperienceLevel extends PlayerStateProperty[Integer]:
  override def getConfigName: String = "experience_level"

  override def getDefaultSerializedProperty: Integer = 0

  override def getSerializedPropertyFromPlayer(player: Player): Integer =
    player.getLevel

  override def restoreProperty(player: Player, level: Integer): Unit =
    player.setLevel(level)